### PR TITLE
Simulation update funcs

### PR DESCRIPTION
**Pull request by rjplevin**
- Added methods to manipulate a `Simulation` instance
- Eliminated `sim.dist_rvs`
- Changed uses of `sim.dist_rvs` to call `values(sim.rvdict)`

See also my comments in #474

### CODE REVIEW

**Comment by rjplevin** [comment]
These new fns are not used in the code base. I though we might merge them to see if they meet our needs.

**Comment by ckingdon95** [comment]
These functions look great to me! I'm looking at the Travis build though, should we be concerned that something is erroring when trying to do an LHS sample?

File: test/mcs/test_defmcs.jl [hunk 1]
```diff
@@ -11,7 +11,6 @@ using Test
 using Mimi: reset_compdefs, modelinstance, compinstance, 
             get_var_value, OUTER, INNER, ReshapedDistribution
 
-reset_compdefs()
```
Comment (ckingdon95) [inline test/mcs/test_defmcs.jl]: Why is this removed? (I'm not sure I understand why it was there in the first place, but just wondering what's different now)

File: src/mcs/lhs.jl [hunk 2]
```diff
@@ -169,7 +169,7 @@ function lhs!(sim::LatinHypercubeSimulation; corrmatrix::Union{Matrix{Float64},N
     trials = sim.trials
     rvdict = sim.rvdict
     num_rvs = length(rvdict)
-    rvlist = sim.dist_rvs
+    rvlist = collect(values(sim.rvdict))
```
Comment (ckingdon95) [inline src/mcs/lhs.jl]: I just add this extra `collect` because the `lhs` function that gets called below iexpects a vector of RVs, not an iterator. All of the other places where you replace `sim.dist_rvs` with `values(sim.rvdict)` work though because they only get used in for loops

File: test/mcs/test_defmcs.jl (hunk 1 above)
Comment (rjplevin) [reply test/mcs/test_defmcs.jl]: This follows from the removal of the global list of component defs. Nothing to reset!

**Comment by davidanthoff** [comment]
I do think we should get back into the grove of not merging things that break CI. Can we figure out what is happening there before we merge?

**Comment by ckingdon95** [comment]
@davidanthoff yes there is an actual error, I fixed one of them but there's still another so we definitely should not merge this yet!

**Comment by ckingdon95** [comment]
I just added another spot where we need to use `collect`, but there are now more errors in something to do with GSA. @rjplevin can you take a look? this might involve needing to update something in the GlobalSensitivityAnalysis package, I'm not sure. The error happens during this call to sample on line 32 of  "src/mcs/sobol.jl":
```
    payload = create_GSA_payload(sim)
    samples = GlobalSensitivityAnalysis.sample(payload)
```

**Comment by rjplevin** [comment]
I just posted an update to this branch that passes all Mimi tests.

**Comment by davidanthoff** [comment]
Ready to merge then? @rjplevin do you want to merge, you probably know best whether this is ready?

**Comment by rjplevin** [comment]
I'm testing against all the models... I'll merge it when everything passes!

**Comment by rjplevin** [comment]
All Mimi and model tests are now passing.
